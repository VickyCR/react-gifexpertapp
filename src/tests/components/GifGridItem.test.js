import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from './../../components/GifGridItem';

describe('GifGridItem component', () => {
  const title = 'Title of the image';
  const url = 'http://domain.com/image.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should be created', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a paragraph with the title', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title)
  });

  test('should have an image with url and alt', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should have class animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');

    expect(className.includes('animate__fadeIn')).toBeTruthy();
  });


});
