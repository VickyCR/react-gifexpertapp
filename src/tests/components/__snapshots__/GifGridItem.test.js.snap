// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem component should be created 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title of the image"
    src="http://domain.com/image.jpg"
  />
  <p>
    Title of the image
  </p>
</div>
`;
